fs2-google-pubsub-http/src/main/scala/com/permutive/pubsub/http/util/RefreshableEffect.scala

```diff
@@ -0,0 +1,109 @@
+package com.permutive.pubsub.http.util
+
+import cats.MonadError
+import cats.effect.concurrent.Ref
+import cats.effect.syntax.concurrent._
+import cats.effect.{Concurrent, Resource, Sync, Timer}
+import cats.syntax.applicativeError._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import fs2.Stream
+
+import scala.concurrent.duration.FiniteDuration
+
+/**
+  * Represents a value of type `A` with effects in `F` which is refreshed.
+  *
+  * Refreshing can be cancelled by evaluating `cancelToken`.
+  *
+  * Implementation is backed by a `cats-effect` `Ref` so evaluating the value is fast.
+  *
+  */
+final class RefreshableEffect[F[_], A] private (val value: F[A], val cancelToken: F[Unit])
```

Comment:
Not a big deal, but what do you think about using the `cats.effect.CancelToken` alias here to clarify the intent?

Reply:
Ah nice, I do like it.